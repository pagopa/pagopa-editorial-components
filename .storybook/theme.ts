// @ts-nocheck
import { createTheme } from '@mui/material/styles';
import { theme } from '@pagopa/mui-italia';

type CustomTheme = {
  [Key in keyof typeof theme]: (typeof theme)[Key];
};

declare module '@mui/material/styles/createTheme' {
  interface Theme extends CustomTheme {}
  interface ThemeOptions extends CustomTheme {}
};

export default createTheme({ ...theme });
