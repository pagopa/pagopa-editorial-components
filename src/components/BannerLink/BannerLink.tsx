import { alpha, Box, Container, Stack, useTheme } from '@mui/material';
import { Ctas, type CtaProps } from '../../components/Ctas';
import { type Generic, type CommonProps } from 'types/components';
import { type BannerLinkContentProps, Content } from './Content';
import { isJSX } from '../../utils';

export interface BannerLinkProps
  extends CommonProps,
    BannerLinkContentProps,
    CtaProps {
  decoration?: string | Generic;
}

export const BannerLink = ({
  theme,
  body,
  title,
  ctaButtons,
  decoration = <></>,
}: BannerLinkProps) => {
  const { palette } = useTheme();

  const backgroundColor =
    theme === 'dark' ? palette.primary.dark : alpha(palette.info.main, 0.08);

  return (
    <Box bgcolor={backgroundColor}>
      <Container>
        <Stack gap={4} sx={styles.main}>
          {isJSX(decoration) ? (
            decoration
          ) : (
            <img src={decoration} alt="Banner Icon" />
          )}
          <BannerLink.Content {...{ body, title, theme }} />
          {ctaButtons?.length && (
            <Ctas
              theme={theme}
              ctaButtons={ctaButtons}
              sx={{
                flexDirection: { xs: 'column', md: 'row' },
              }}
            />
          )}
        </Stack>
      </Container>
    </Box>
  );
};

BannerLink.Content = Content;

const styles = {
  main: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: { md: '64px 24px', xs: '32px 24px' },
  },
};
