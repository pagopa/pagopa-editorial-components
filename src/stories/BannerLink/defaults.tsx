import { type BannerLinkProps } from 'components/BannerLink';

export const defaults: {
  args: BannerLinkProps;
  ctaButtons: BannerLinkProps['ctaButtons'];
  body: BannerLinkProps['body'];
} = {
  args: {
    theme: 'light',
    title: 'Vuoi diventare un Partner tecnologico qualificato?',
    body: 'Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it',
    visibleIcon: false,
  },
  body: (
    <p>
      Leggi le domande frequenti sull’adesione a <b>pagoPA</b> da parte degli
      Enti Creditori oppure scrivi a:{' '}
      <a href="mailto:account@pagopa.it">account@pagopa.it</a>
    </p>
  ),
  ctaButtons: [
    {
      text: 'LEGGI LE FAQ',
      onClick: () => {
        alert('LEGGI LE FAQ');
      },
    },
    {
      text: 'LEGGI LE FAQ',
      onClick: () => {
        alert('LEGGI LE FAQ');
      },
    },
  ],
};
