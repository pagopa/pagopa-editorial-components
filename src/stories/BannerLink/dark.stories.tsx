import { type ComponentMeta, type ComponentStory } from '@storybook/react';
import { BannerLink } from '../../components/BannerLink';
import { defaults } from './defaults';
import LoginImg from '../assets/svg/Login';

export default {
  component: BannerLink,
  title: 'BannerLink/dark',
  argTypes: {},
  args: {
    ...defaults.args,
    theme: 'dark',
  },
} as ComponentMeta<typeof BannerLink>;

const Template: ComponentStory<typeof BannerLink> = (args) => (
  <BannerLink {...args} />
);

export const Standard = Template.bind({});

export const OneButton = Template.bind({});
OneButton.args = {
  ctaButtons: [
    {
      color: 'negative',
      variant: 'outlined',
      text: 'LEGGI LE FAQ',
      onClick: () => {
        alert('LEGGI LE FAQ');
      },
    },
  ],
};

export const TwoButtons = Template.bind({});
TwoButtons.args = {
  ctaButtons: defaults.ctaButtons,
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  decoration: <LoginImg fill="white" />,
};

export const TwoButtonsAndIcon = Template.bind({});
TwoButtonsAndIcon.args = {
  decoration: <LoginImg fill="white" />,
  ctaButtons: defaults.ctaButtons,
};

export const BodyComponent = Template.bind({});
BodyComponent.args = {
  body: defaults.body,
  ctaButtons: defaults.ctaButtons,
};
