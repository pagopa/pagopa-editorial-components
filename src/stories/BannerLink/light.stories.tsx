import { type ComponentMeta, type ComponentStory } from '@storybook/react';
import { BannerLink } from '../../components/BannerLink';
import { defaults } from './defaults';
import LoginImg from '../assets/svg/Login';

export default {
  component: BannerLink,
  title: 'BannerLink/light',
  argTypes: {},
  args: defaults.args,
} as ComponentMeta<typeof BannerLink>;

const Template: ComponentStory<typeof BannerLink> = (args) => (
  <BannerLink {...args} />
);

export const Standard = Template.bind({});

export const OneButton = Template.bind({});
OneButton.args = {
  ctaButtons: [
    {
      color: 'primary',
      variant: 'outlined',
      text: 'LEGGI LE FAQ',
      onClick: () => {
        alert('LEGGI LE FAQ');
      },
    },
  ],
};

export const TwoButtons = Template.bind({});
TwoButtons.args = {
  ctaButtons: defaults.ctaButtons,
};

export const WithIcon = Template.bind({});
WithIcon.args = {
  decoration: <LoginImg fill="#17324d" />,
};

export const TwoButtonsAndIcon = Template.bind({});
TwoButtonsAndIcon.args = {
  decoration: <LoginImg fill="#17324d" />,
  ctaButtons: defaults.ctaButtons,
};

export const BodyComponent = Template.bind({});
BodyComponent.args = {
  body: defaults.body,
  ctaButtons: defaults.ctaButtons,
};
