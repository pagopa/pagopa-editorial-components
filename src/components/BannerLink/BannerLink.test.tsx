import { BannerLink } from './BannerLink';
import '@testing-library/jest-dom';
import { fireEvent, render } from '@testing-library/react';
import { createTheme, ThemeProvider } from '@mui/material';
import { theme } from '../../../.storybook/theme';

/**
 * @jest-environment jsdom
 */

import renderer from 'react-test-renderer';

const mainTheme = createTheme({
  ...theme,
});

it('renders correctly standard', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly one button', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        ctaButtons={[
          {
            text: 'LEGGI LE FAQ',
            onClick: () => {},
            color: 'primary',
          },
        ]}
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly two buttons', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        ctaButtons={[
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
        ]}
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with icon', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        decoration={{
          alt: 'Login',
          src: 'static/media/login-white.9e86b706.svg',
        }}
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with icon and two buttons', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        ctaButtons={[
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
        ]}
        decoration={{
          alt: 'Login',
          src: 'static/media/login-white.9e86b706.svg',
        }}
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with body ', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        ctaButtons={[
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
        ]}
        decoration={{
          alt: 'Login',
          src: 'static/media/login-white.9e86b706.svg',
        }}
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with custom decorator', () => {
  const tree = renderer
    .create(
      <BannerLink
        body={
          <p>
            Leggi le domande frequenti sull’adesione a <b>pagoPA</b> da parte
            degli Enti Creditori oppure scrivi a:{' '}
            <a href="mailto:account@pagopa.it">account@pagopa.it</a>
          </p>
        }
        ctaButtons={[
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
          {
            onClick: () => {},
            text: 'LEGGI LE FAQ',
          },
        ]}
        decoration={
          <div
            style={{
              alignItems: 'center',
              backgroundColor: 'white',
              borderRadius: '50%',
              display: 'flex',
              fontSize: '32px',
              height: '100px',
              justifyContent: 'center',
              width: '100px',
            }}
          >
            PA
          </div>
        }
        theme="light"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly standard dark', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        theme="dark"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly one button dark', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        ctaButtons={[
          {
            text: 'LEGGI LE FAQ',
            onClick: () => {},
            color: 'primary',
          },
        ]}
        theme="dark"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with icon dark', () => {
  const tree = renderer
    .create(
      <BannerLink
        body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
        decoration={{
          alt: 'Login',
          src: 'static/media/login-white.9e86b706.svg',
        }}
        theme="dark"
        title="Vuoi diventare un Partner tecnologico qualificato?"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly two buttons dark', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={mainTheme}>
        <BannerLink
          body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
          ctaButtons={[
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
          ]}
          theme="dark"
          title="Vuoi diventare un Partner tecnologico qualificato?"
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with icon and two buttons dark', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={mainTheme}>
        <BannerLink
          body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
          ctaButtons={[
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
          ]}
          decoration={{
            alt: 'Login',
            src: 'static/media/login-white.9e86b706.svg',
          }}
          theme="dark"
          title="Vuoi diventare un Partner tecnologico qualificato?"
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with body dark ', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={mainTheme}>
        <BannerLink
          body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
          ctaButtons={[
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
          ]}
          decoration={{
            alt: 'Login',
            src: 'static/media/login-white.9e86b706.svg',
          }}
          theme="dark"
          title="Vuoi diventare un Partner tecnologico qualificato?"
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with custom decorator dark', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={mainTheme}>
        <BannerLink
          body={
            <p>
              Leggi le domande frequenti sull’adesione a <b>pagoPA</b> da parte
              degli Enti Creditori oppure scrivi a:{' '}
              <a href="mailto:account@pagopa.it">account@pagopa.it</a>
            </p>
          }
          ctaButtons={[
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
            {
              onClick: () => {},
              text: 'LEGGI LE FAQ',
            },
          ]}
          decoration={
            <div
              style={{
                alignItems: 'center',
                backgroundColor: 'white',
                borderRadius: '50%',
                display: 'flex',
                fontSize: '32px',
                height: '100px',
                justifyContent: 'center',
                width: '100px',
              }}
            >
              PA
            </div>
          }
          theme="dark"
          title="Vuoi diventare un Partner tecnologico qualificato?"
        />
      </ThemeProvider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders text correctly', async () => {
  const { getByText } = render(
    <BannerLink
      body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
      ctaButtons={[
        {
          text: 'LEGGI LE FAQ',
          onClick: () => {},
          color: 'primary',
        },
      ]}
      theme="light"
      title="Vuoi diventare un Partner tecnologico qualificato?"
    />
  );
  expect(getByText('LEGGI LE FAQ')).toBeInTheDocument();
  expect(
    getByText(
      'Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it'
    )
  ).toBeInTheDocument();
  expect(
    getByText('Vuoi diventare un Partner tecnologico qualificato?')
  ).toBeInTheDocument();
});

test('alert is fired correctly', async () => {
  const alertMock = jest.spyOn(window, 'alert').mockImplementation();

  const { getByText } = render(
    <BannerLink
      body="Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it"
      ctaButtons={[
        {
          text: 'LEGGI LE FAQ',
          onClick: () => {
            alert('test');
          },
          color: 'primary',
        },
      ]}
      theme="light"
      title="Vuoi diventare un Partner tecnologico qualificato?"
    />
  );
  fireEvent.click(getByText('LEGGI LE FAQ'));
  expect(alertMock).toHaveBeenCalledTimes(1);
});
