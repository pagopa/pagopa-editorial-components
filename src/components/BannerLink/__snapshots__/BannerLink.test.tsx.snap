// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly one button 1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it
        </p>
      </div>
      <div
        className="MuiStack-root css-v53bf6-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1desrfe-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly standard 1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly two buttons 1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it
        </p>
      </div>
      <div
        className="MuiStack-root css-v53bf6-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1desrfe-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1olp7er-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with body  1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <img
        alt="Login"
        src="static/media/login-white.9e86b706.svg"
      />
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it
        </p>
      </div>
      <div
        className="MuiStack-root css-v53bf6-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1desrfe-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1olp7er-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with custom decorator 1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <div
        style={
          {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": "50%",
            "display": "flex",
            "fontSize": "32px",
            "height": "100px",
            "justifyContent": "center",
            "width": "100px",
          }
        }
      >
        PA
      </div>
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          <p>
            Leggi le domande frequenti sull’adesione a 
            <b>
              pagoPA
            </b>
             da parte degli Enti Creditori oppure scrivi a:
             
            <a
              href="mailto:account@pagopa.it"
            >
              account@pagopa.it
            </a>
          </p>
        </p>
      </div>
      <div
        className="MuiStack-root css-v53bf6-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1desrfe-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1olp7er-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
        </button>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with icon 1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <img
        alt="Login"
        src="static/media/login-white.9e86b706.svg"
      />
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it
        </p>
      </div>
    </div>
  </div>
</section>
`;

exports[`renders correctly with icon and two buttons 1`] = `
<section
  className="MuiBox-root css-14prgyt"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      className="MuiStack-root css-1ex9eoo-MuiStack-root"
    >
      <img
        alt="Login"
        src="static/media/login-white.9e86b706.svg"
      />
      <div
        className="MuiStack-root css-54p0y9-MuiStack-root"
      >
        <h6
          className="MuiTypography-root MuiTypography-h6 css-1uaewpj-MuiTypography-root"
        >
          Vuoi diventare un Partner tecnologico qualificato?
        </h6>
        <p
          className="MuiTypography-root MuiTypography-body2 css-1yr8029-MuiTypography-root"
        >
          Leggi le domande frequenti sull’adesione a pagoPA da parte degli Enti Creditori oppure scrivi a: account@pagopa.it
        </p>
      </div>
      <div
        className="MuiStack-root css-v53bf6-MuiStack-root"
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1desrfe-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
        </button>
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1olp7er-MuiButtonBase-root-MuiButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          text="LEGGI LE FAQ"
          type="button"
        >
          LEGGI LE FAQ
        </button>
      </div>
    </div>
  </div>
</section>
`;
